### PR TITLE
Update extension

### PR DESCRIPTION
**Pull request by weiglszonja**
Update `Miniscope`
- Update extension spec based on the example data (`Miniscope V3`)  received from Tye lab (new attributes from #2)
- Add tests

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 setup_args = {
     "name": "ndx-miniscope",
-    "version": "0.3.0",
+    "version": "0.3.1",
```
Comment (weiglszonja) [inline setup.py:19]: This is bigger than a patch
```suggestion
    "version": "0.4.0",
```

File: src/pynwb/ndx_miniscope/miniscope.py [hunk 2]
```diff
@@ -1,18 +1,103 @@
 import os
 
 import pandas as pd
+from hdmf import docval
+from hdmf.utils import get_docval, popargs
+from pynwb.device import Device
 from pynwb.misc import AnnotationSeries
 
-from pynwb import get_class, load_namespaces
```
Comment (weiglszonja) [inline src/pynwb/ndx_miniscope/miniscope.py:6]: if `get_class` works than this is not needed

File: src/spec/create_extension_spec.py [hunk 3]
```diff
@@ -6,18 +6,68 @@ def main():
     ns_builder = NWBNamespaceBuilder(
         doc="holds metadata relevant for miniscope acquisition",
         name="ndx-miniscope",
-        version="0.2.2",
+        version="0.3.1",
```
Comment (weiglszonja) [inline src/spec/create_extension_spec.py:9]: modify version here too

Comment (weiglszonja) [inline src/spec/create_extension_spec.py:9]:
```suggestion
        version="0.4.0",
```

File: spec/ndx-miniscope.extensions.yaml [hunk 4]
```diff
@@ -1,13 +1,49 @@
 groups:
 - neurodata_type_def: Miniscope
   neurodata_type_inc: Device
-  doc: extension of Device to hold metadata specific to Miniscopes
+  doc: The extension of Device to hold metadata specific to Miniscopes.
   attributes:
+  - name: version
+    dtype: text
+    doc: The version of Miniscope e.g., V3, V4.
+    required: false
```
[the remark this answers is not in the document]
Comment (CodyCBakerPhD) [reply spec/ndx-miniscope.extensions.yaml:9]: Also, would this information not be contained in the `deviceType`?

[the remark this answers is not in the document]
Comment (weiglszonja) [reply spec/ndx-miniscope.extensions.yaml:9]: You're right, according to [this](https://github.com/Aharoni-Lab/Miniscope-DAQ-QT-Software/wiki/Device-Configuration-Files), `deviceType` will contain `V3` or `V4` suffix. In that case I'll remove the `version` field.

**Comment by CodyCBakerPhD** [comment]
@weiglszonja This looks good to me; merge when you're ready

**Pull request merged**